pkg/system/phase4_configuring.go

```diff
@@ -1523,3 +1529,142 @@ func (r *Reconciler) ReconcileNamespaceStores(namespaceResources []nb.NamespaceR
 	}
 	return nil
 }
+
+// ReconcileNoobaaDeletionPolicy reconciles the deletion policy of a Noobaa resource
+// and configures the noobaa admission controller appropriately.
+func (r *Reconciler) ReconcileNoobaaDeletionPolicy() error {
+	noobaaAdmissionEnabled := os.Getenv("ENABLE_NOOBAA_ADMISSION")
+	if strings.ToLower(noobaaAdmissionEnabled) != "true" {
+		return nil
+	}
+
+	rule := GetNoobaaCRDeletionAdmissionRule()
+
+	if r.NooBaa.Spec.CleanupPolicy.AllowNoobaaDeletion {
+		return RemoveRuleFromNoobaaAdmissionWebhook(rule)
+	}
+
+	return AddRuleToNoobaaAdmissionWebhook(rule)
+}
+
+// RemoveRuleFromNoobaaAdmissionWebhook removes the rule from the noobaa internal admission webhook
+// if it exists
+func RemoveRuleFromNoobaaAdmissionWebhook(rule *admissionv1.RuleWithOperations) error {
+	wc := &admissionv1.ValidatingWebhookConfiguration{}
+	wc.SetName("admission-validation-webhook")
+
+	// Find the pre-existing admission webhook
+	if _, _, err := util.KubeGet(wc); err != nil {
+		return fmt.Errorf("failed to get webhook: %v", err)
+	}
+
+	// If no rule was removed then return
+	if !removeAdmissionRule(wc, rule) {
+		return nil
+	}
+
+	if !util.KubeUpdate(wc) {
+		return fmt.Errorf("failed to update webhook")
+	}
+
+	return nil
+}
+
+// AddRuleToNoobaaAdmissionWebhook adds the rule to the noobaa admission webhook if it
+// doesn't exists
+func AddRuleToNoobaaAdmissionWebhook(rule *admissionv1.RuleWithOperations) error {
+	wc := &admissionv1.ValidatingWebhookConfiguration{}
+	wc.SetName("admission-validation-webhook")
+
+	// Find the pre-existing admission webhook
+	if _, _, err := util.KubeGet(wc); err != nil {
+		return fmt.Errorf("failed to get webhook: %v", err)
+	}
+
+	// If no rule was added then return
+	if !addAdmissionRule(wc, rule) {
+		return nil
+	}
+
+	if !util.KubeUpdate(wc) {
+		return fmt.Errorf("failed to update webhook")
+	}
+
+	return nil
+}
+
+// removeAdmissionRule removes the matching rule from the given webhook and returns true
+// if rule was deleted or else returns false
+func removeAdmissionRule(wc *admissionv1.ValidatingWebhookConfiguration, rule *admissionv1.RuleWithOperations) bool {
+	ruleIdx, found := findAdmissionRule(wc, rule)
+	if !found {
+		return false
+	}
+
+	wc.Webhooks[0].Rules = append(
+		wc.Webhooks[0].Rules[:ruleIdx],
+		wc.Webhooks[0].Rules[ruleIdx+1:]...,
+	)
+
+	return true
+}
+
+// addAdmissionRule takes in pointer to validationwebhookconfiguration and adds the given rule to it and returns true
+// if rule was added or else returns false
+//
+// NOTE: It will not add the rule if it exists
+func addAdmissionRule(wc *admissionv1.ValidatingWebhookConfiguration, rule *admissionv1.RuleWithOperations) bool {
+	_, found := findAdmissionRule(wc, rule)
+	if found {
+		return false
+	}
+
+	wc.Webhooks[0].Rules = append(wc.Webhooks[0].Rules, *rule)
```

Comment:
@utkarsh-pro don't you need to `KubeUpdate(wc)` after appending a new rule?
or after any change done to it?